### ISSUE TITLE
[Question] Make GridSLiM header take up all the columns rather than just one

### ISSUE THREAD

**Issue by BrandonSlaght** [TonicArtos/SuperSLiM#159]
Hello,
I was a little confused by the documentation on this project for grid layouts.  I have a header.xml file:

```
<TextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:text="Albums"
    app:slm_headerDisplay="inline|sticky"
    app:slm_isHeader="true"
    app:slm_section_headerMarginStart="match_header"
    xmlns:android="http://schemas.android.com/apk/res/android"/>
```
And in my adapter: 
```
public albumholder onCreateViewHolder (ViewGroup p, int type) {
        if (type == 0) {
            return new albumholder(LayoutInflater.from(p.getContext()).inflate(R.layout.header, p, false), c);
        } else {
            return new albumholder(LayoutInflater.from(p.getContext()).inflate(R.layout.result, p, false), c);
        }
    }
...
public void onBindViewHolder(albumholder h, int position) {
        h.bind(results.get(position));
        final GridSLM.LayoutParams params = new GridSLM.LayoutParams(h.getLayoutParams());
        params.setSlm(GridSLM.ID);
        params.setFirstPosition(0);
        h.setLayoutParams(params);
    }

    @Override
    public int getItemViewType(int position) {
        return results.get(position).isHeader ? 0 : 1; //0 for head, 1 for album
    }
```

however, what I get is this: 
![image](https://cloud.githubusercontent.com/assets/7023112/12069162/dc686268-afef-11e5-9287-df47ece92f21.png)

The header occupies a grid space.  Instead, I want it to take up a bar across both columns in the grid.  What options do I use to achieve this?

**Comment by muetzenflo** [comment]
Hi, I had the same problem and after a lot(!) of try&error I found out that you must not use ViewGroups in the header xml. My header layout wraped to TextViews in a LinearLayout. I merged these TextViews together and got rid of the LinearLayout and all of a sudden it worked.

```
<!-- header.xml -->
<?xml version="1.0" encoding="utf-8"?>
<TextView
    android:id="@+id/tv_headline_title"
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    app:slm_headerDisplay="inline|sticky"
    app:slm_isHeader="true"
    tools:text="Header Item"
/>
```

I did not test if it is the same with the content.xml since I only need a single ImageView in there (for now).

@TonicArtos Is it possible to use "complex" layouts for the header and/or the content XMLs? Even a FrameLayout does not work right now (for example to view an ImageView with a badge in one corner).

**Comment by BrandonSlaght** [comment]
Thanks for the reply, glad you got yours working.  However, I do not have any viewgroups in my header.  Can I see some of your elated code? I'm STILL trying to get this working.

**Comment by infinity8942** [comment]
@BrandonSlaght I have the same problem.Looks like the property "inline|sticky" doesn't work.

**Comment by BrandonSlaght** [comment]
@muetzenflo mentioned they managed to get something working, but I haven't heard back from them or @TonicArtos, which makes sense because he hasn't been on GitHub since november :(

**Comment by infinity8942** [comment]
@BrandonSlaght ok...I made a mistake using "GridLayoutManager" instead of "LayoutManager".Now it worked. Hopefully it will be of use to you.

**Comment by TonicArtos** [comment]
Sorry about things. I'll have a good news report fairly soon.

**Comment by BrandonSlaght** [comment]
I managed to debug this problem.  It was stemming from the fact that I was using this recyclerview with a library called MaterialViewPager, and when registering the adapter with the materialviewpager, it discarded all the extra information that superSLiM needs.  I ended up extending the materialviewpager adapter with my own class which supported both the SLiM parameters and MVP parameters that both classes needed.  I can post a gist if anyone has the same issue with that library but I'm a bit embarrassed about the code quality :dizzy_face: Anyways, for now, this issue is closed for me.
